core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -668,6 +668,18 @@ class ParserTest extends munit.ScalaCheckSuite {
     parseTest(Parser.stringIn(List("foo", "foobar", "foofoo", "foobat")).string, "foobal", "foo")
   }
 
+  test("Parser.careted and spanned works") {
```

Comment:
also, I'd like to see a law like `pa.parse(str) == pa.withSpan.map(_._1).parse(str)` and `pa.spanOf.parse(str) == pa.withSpan.map(_._2).parse(str)`

something like that.